CMakeLists.txt

```diff
@@ -133,6 +128,29 @@ if(GIT_EXECUTABLE)
         -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/GenerateVersionHeader.cmake)
 endif()
 
+add_compile_definitions(SPDLOG_FMT_EXTERNAL)
+add_compile_definitions(FMT_HEADER_ONLY)
+add_compile_definitions(GTEST_USE_STATIC_LIBS)
+
+find_package(nlohmann_json REQUIRED)
+find_package(spdlog_setup REQUIRED)
+find_package(spdlog REQUIRED)
+find_package(fmt REQUIRED)
+find_package(cpptoml REQUIRED)
+
+include_directories(${nlohmann_json_INCLUDE_DIR})
+include_directories(${spdlog_INCLUDE_DIR})
+include_directories(${spdlog_setup_INCLUDE_DIR})
+include_directories(${fmt_INCLUDE_DIR})
+include_directories(${cpptoml_INCLUDE_DIR})
+
+set(CONAN_LIBS 
+    nlohmann_json::nlohmann_json 
+    spdlog::spdlog
+    spdlog_setup::spdlog_setup
+    fmt::fmt
+)
```

Comment:
While not wrong for a library consisting of only a small number of components, it's usually preferable to do `target_link_libraries(<target> PUBLIC/PRIVATE/INTERFACE nlohmann_json::nlohmann_json ...)` to apply the dependency info directly to the relevant target (which is `common` in this case) and have CMake propagate the dependency automatically. PUBLIC/PRIVATE should be based on whether the dependency is used in the public headers of the component or not.

Reply:
done!